components/org.apache.stratos.cloud.controller/src/main/java/org/apache/stratos/cloud/controller/context/CloudControllerContext.java

```diff
@@ -729,4 +740,97 @@ private void copyList(List sourceList, List destinationList) {
             destinationList.add(item);
         }
     }
+
+    public Map<String, Map<String, IaasProvider>> getPartitionToIaasProviderByCartridge() {
+        return partitionToIaasProviderByCartridge;
+    }
+
+    public void setPartitionToIaasProvider(Map<String, Map<String, IaasProvider>> partitionToIaasProviderByCartridge) {
+        this.partitionToIaasProviderByCartridge = partitionToIaasProviderByCartridge;
+    }
+
+    public void addIaasProvider(String cartridgeType, String partitionId, IaasProvider iaasProvider) {
+        Map<String, IaasProvider> partitionToIaasProvider = new ConcurrentHashMap<String, IaasProvider>();
+        partitionToIaasProvider.put(partitionId, iaasProvider);
+        this.partitionToIaasProviderByCartridge.put(cartridgeType, partitionToIaasProvider);
+    }
+
+    public void addIaasProviders(String cartridgeType, Map<String, IaasProvider> map) {
```

Comment:
This line would have been much readable if some other meaningful name was used to replace the variable name "map".